### PR TITLE
PYIC-3669: Add CredentialEvidenceItemTest

### PR DESCRIPTION
**Pull request by MikeCollingwood**
### What changed

- Added CredentialEvidenceItemTest

### Why did it change

- There was no test file and these changes were made as part of a reformatting ticket which was abandoned, so this ticket scavenges this test file

### Issue tracking

- [PYIC-3669](https://govukverify.atlassian.net/browse/PYIC-3669)

### Environment variables or secrets

- [ ] No environment variables or secrets were added or changed

[PYIC-3669]: https://govukverify.atlassian.net/browse/PYIC-3669?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

**Comment by MikeCollingwood** [comment]
NB: Reaches 92% coverage locally, with branch coverage

File: libs/gpg45-evaluator/src/test/java/uk/gov/di/ipv/core/library/gpg45/domain/CredentialEvidenceItemTest.java [hunk 1]
```diff
@@ -0,0 +1,402 @@
...
+import uk.gov.di.ipv.core.library.gpg45.Gpg45Scores;
+import uk.gov.di.ipv.core.library.gpg45.domain.CredentialEvidenceItem.EvidenceType;
+import uk.gov.di.ipv.core.library.gpg45.exception.UnknownEvidenceTypeException;
+
+import java.util.Arrays;
+import java.util.Collections;
+import java.util.List;
+import java.util.stream.Stream;
+
+import static org.junit.jupiter.api.Assertions.*;
```
Comment (TalatAydinGDS) [inline libs/gpg45-evaluator/src/test/java/uk/gov/di/ipv/core/library/gpg45/domain/CredentialEvidenceItemTest.java]: We don't use '*' and add only needs dependency.

File: libs/gpg45-evaluator/src/test/java/uk/gov/di/ipv/core/library/gpg45/domain/CredentialEvidenceItemTest.java [hunk 2]
```diff
@@ -0,0 +1,402 @@
...
+import static org.junit.jupiter.api.Assertions.assertEquals;
+
+class CredentialEvidenceItemTest {
+    @Test
+    void shouldConstructorWithEvidenceTypeAndScore() {
+        CredentialEvidenceItem credentialEvidenceItem1 =
```
Comment (TalatAydinGDS) [inline libs/gpg45-evaluator/src/test/java/uk/gov/di/ipv/core/library/gpg45/domain/CredentialEvidenceItemTest.java]: I think, you can create a method or can use ParameterizedTest here. There are so many duplicate code here.

File: libs/gpg45-evaluator/src/test/java/uk/gov/di/ipv/core/library/gpg45/domain/CredentialEvidenceItemTest.java [hunk 3]
```diff
@@ -0,0 +1,402 @@
...
+        assertEquals(10, credentialEvidenceItem.getStrengthScore());
+        assertEquals(20, credentialEvidenceItem.getValidityScore());
+        assertEquals(30, credentialEvidenceItem.getActivityHistoryScore());
+        assertEquals(40, credentialEvidenceItem.getVerificationScore());
+    }
+
+    @Test
+    void shouldGetTypeActivity() throws UnknownEvidenceTypeException {
+        CredentialEvidenceItem credentialEvidenceItem =
+                new CredentialEvidenceItem(EvidenceType.ACTIVITY, 30, Collections.emptyList());
```
Comment (TalatAydinGDS) [inline libs/gpg45-evaluator/src/test/java/uk/gov/di/ipv/core/library/gpg45/domain/CredentialEvidenceItemTest.java]: I think, can use builder()

File: libs/gpg45-evaluator/src/test/java/uk/gov/di/ipv/core/library/gpg45/domain/CredentialEvidenceItemTest.java [hunk 4]
```diff
@@ -0,0 +1,402 @@
...
+    @Test
+    void shouldGetEvidenceScore() {
+        CredentialEvidenceItem credentialEvidenceItem =
+                CredentialEvidenceItem.builder().strengthScore(10).validityScore(20).build();
+        assertEquals(new Gpg45Scores.Evidence(10, 20), credentialEvidenceItem.getEvidenceScore());
+    }
+
+    @Test
+    void shouldHasContraIndicators() {
+        CredentialEvidenceItem credentialEvidenceItem1 = new CredentialEvidenceItem(10, 20, null);
```
Comment (TalatAydinGDS) [inline libs/gpg45-evaluator/src/test/java/uk/gov/di/ipv/core/library/gpg45/domain/CredentialEvidenceItemTest.java]: I think, if we are not writing constructor test, I think using a builder() would be a great choice.

**Review by TalatAydinGDS — CHANGES_REQUESTED** [review]
I cannot see code coverage and Sonarqube report on GitHub. Please ask to SRE team, there might be an issue. We need to see what review code coverage and Sonar feedback.

**Review by TalatAydinGDS — APPROVED** [review]
LGTM

**Pull request merged**